src/box/raft.c

```diff
@@ -657,6 +683,12 @@ box_raft_init(void)
 		       NULL, NULL);
 	trigger_create(&box_raft_on_quorum_loss, box_raft_on_quorum_change_f,
 		       NULL, NULL);
+	struct watcher *watcher = xmalloc(sizeof(*watcher));
+	const char *key = "box.wal_error";
+	size_t key_len = strlen(key);
+	box_register_watcher(key, key_len, box_raft_on_wal_error_f,
+			     (watcher_destroy_f)free, 0,
+			     watcher);
```

Comment:
Why do you need a watcher for this and not some sort of journal's internal `struct trigger` member? Watchers are public. Why is this event needed in public?

If there is a justification, then the WAL error event introduction should be in a separate commit with its own test, and the new event needs to be documented (AFAIK, we do document public events, do we?).

Reply:
The WAL error event was already entered in https://github.com/tarantool/tarantool/pull/9422 and was also documented.
Of course, it would be possible to use a trigger here, however it seems that using wather has more advantages than disadvantages.
disadvantages of watchers compared to triggers:
* delivering messages in a separate fiber
* serialize data before delivering
However, the `watchable_node` "box.wal_error" already exists. That is, for every event there, the `watchable->worker` fiber already wakes up regardless of this patch. As for data serialization, in our case there is only one `uint64_t`. Moreover, data serialization occurs regardless of whether there are subscribed wathcers on watchable node. That is, adding a subscriber does not increase the number of serializations. I do not use the data itself in the handler, that is, I do not deserialize it.

advantages:
* We don't make the code too difficult to read. It will look strange if we call the trigger everywhere we call `broadcast_fmt("wal.error", ...`.

But I don’t know how important it is for us to comply with the accepted convention that we do not use watchers in system code. Is this a strict agreement?))